GestioneOrari/src/Elaborato/ProcessedFile.java

```diff
@@ -0,0 +1,23 @@
+package Elaborato;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import row.Row;
+
+public class ProcessedFile {
+	private List<Row> rows;
+	
+	public ProcessedFile() {
+		rows = new ArrayList<>();
+	}
+	
+	public void addRowToRows(Date date, String startHour, String endHour, String order, String note) {
```

Comment:
Non ricordo se nel materiale c'è qualche riferimento al dependency injection, ma in questo caso è chiaro che la richiesta di così tanti parametri è un po' eccessiva per quanto giusta. Qui ci starebbe proprio bene utilizzare quel design pattern. C'è un articolo molto interessante su medium, se vuoi leggerlo lo trovi qui: [What Is Dependency Injection?](https://medium.com/better-programming/what-is-dependency-injection-b2671b1ea90a), che dice:

> "In your code, the interaction between objects should be as thin and clean as possible. You should ask for precisely what you need and nothing more. If you need something, you should ask for it — have it “delivered” to you instead of trying to create it yourself." 

Non so se l'hai mai usato questo design, ma rende tutto più indipendente e pulito.

In questo caso dovresti sostituire gli argomenti del metodo con l'oggetto che vuoi inserire:

```suggestion
	public void addRowToRows(Row row) {
```

Questo design è consigliato nel caso in cui all'interno del metodo chiamato c'è una creazione di un oggetto dipendente dagli argomenti del metodo stesso.

> "Dependency injection is a twenty-five dollar term for a ten-cent concept: If you need something, ask for it. Don’t create things yourself — push that off to someone else. Every class of any substance will likely need help from other classes. If your class needs help, ask for it; don’t try to “do it yourself.” Remember, every time you try to “do it yourself,” you create a hard-coded dependency. And again, those should be avoided like global variables."

@scrivereleonardo @MariusLovesPizza @ebrogini considerate anche voi questo commento che è importante.

Reply:
Grazie per la spiegazione, conoscevo il pattern ma non mi sono reso conto fosse applicabile. Ora ho le idee più chiare.